Error 404 on leaflet.py example

Hi
I created a web-app based on the leaflet.py example. It has worked fine for several years but after a few month since I did not use (Jan 2023),  I get a "urllib.error.HTTPError: HTTP Error 404: Not Found".

So, I got back to the leaflet.py code from flexx readthedocs and test it as is --> the example return the same error.

What has change since the beginning of the year that makes this example to stop working ?

Thanks for any help,

Hi, thanks a lot for your help. However I'm still having trouble.
I get this error message from _get data function:

assert res.ok, f"{res.status_code}: {res.reason}"
AssertionError: 404: Not Found

In addition, I'm now having a warning message:

/Users/admin/PycharmProjects/Genealogie/env/lib/python3.7/site-packages/urllib3/__init__.py:38: NotOpenSSLWarning: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with 'LibreSSL 2.8.3'. See: https://github.com/urllib3/urllib3/issues/3020
  exceptions.NotOpenSSLWarning,

Not sure if it is Mac environment or if it is because I'm using Python 3.8.

I've kind of solve the issue of urllib3 following #3020 issue on urllib3.
Still, the leaflet.py example give me this error, even we the suggested modifications:

Traceback (most recent call last):
  File "/Users/admin/PycharmProjects/Genealogie/sample/ui/essai.py", line 82, in <module>
    flx.assets.add_shared_data(icon, _get_data('images/%s' % icon))
  File "/Users/admin/PycharmProjects/Genealogie/sample/ui/essai.py", line 53, in _get_data
    assert res.ok, f"{res.status_code}: {res.reason}"
AssertionError: 404: Not Found

--- a/flexxamples/howtos/leaflet.py
+++ b/flexxamples/howtos/leaflet.py
@@ -9,7 +9,8 @@ import re
 import base64
 import mimetypes
 
-import requests  # Note: you may need to pip install requests
+import requests  # Note: you may need to "pip install requests brotli"
+import brotli  # noqa: requests needs this for br encoding
 from flexx import flx
 
 
@@ -23,7 +24,7 @@ _leaflet_icons = [
 
 request_headers = {
     'User-Agent': 'flexx/%s' % flx.__version__,
-    'Accept-Encoding': 'gzip, deflate'
+    'Accept-Encoding': 'gzip, deflate, br'
 }
 
 if 'LEAFLET_DIR' in os.environ:

--- a/flexxamples/howtos/leaflet.py
+++ b/flexxamples/howtos/leaflet.py
@@ -5,11 +5,11 @@ This example demonstrates the use of Leaflet to display a slippy map.
 
 
 import os
-from urllib.request import urlopen, Request
 import re
 import base64
 import mimetypes
 
+import requests  # Note: you may need to pip install requests
 from flexx import flx
 
 
@@ -21,6 +21,10 @@ _leaflet_icons = [
     'marker-shadow.png',
 ]
 
+request_headers = {
+    'User-Agent': 'flexx/%s' % flx.__version__,
+    'Accept-Encoding': 'gzip, deflate'
+}
 
 if 'LEAFLET_DIR' in os.environ:
     _base_url = 'file://%s' % os.environ['LEAFLET_DIR']
@@ -33,8 +37,9 @@ def _get_code(item):
     """ Get a text item from _base_url
     """
     url = '%s/%s' % (_base_url, item)
-    req = Request(url, headers={'User-Agent': 'flexx/%s' % flx.__version__})
-    return urlopen(req).read().decode()
+    res = requests.get(url , headers=request_headers)
+    assert res.ok, f"{res.status_code}: {res.reason}"
+    return res.text
 
 
 def _get_data(item_or_url):
@@ -44,8 +49,9 @@ def _get_data(item_or_url):
         url = item_or_url
     else:
         url = '%s/%s' % (_base_url, item_or_url)
-    req = Request(url, headers={'User-Agent': 'flexx/%s' % flx.__version__})
-    return urlopen(req).read()
+    res = requests.get(url , headers=request_headers)
+    assert res.ok, f"{res.status_code}: {res.reason}"
+    return res.content
 
 
 def _embed_css_resources(css, types=('.png',)):
@@ -281,5 +287,5 @@ class LeafletExample(flx.Widget):
 
 
 if __name__ == '__main__':
-    flx.launch(LeafletExample, 'firefox')
+    flx.launch(LeafletExample, 'chrome-browser')
     flx.run()

